Get rid of -api in regular expression for SalesforceBulk::Connection.parse_instance()

mahzilla opened the pull request:
Salesforce used to have a -api after the instance name and no longer
does.

Commit:
Get rid of -api in server url

Salesforce used to have a -api after the instance name and no longer
does.

seanpd commented:
Same issue that I mentioned in #15.  I included a monkeypatch workaround.

JamieAyer commented:
Thanks seanpd.  That also worked for me.  It stopped working altogether in Production and QA last week.

tyro89 commented:
Just came to open a pr for the exact same fix but I see you guys beat me to it. Ping @jorgevaldivia as the gem is broken until this is fixed :(

JamieAyer commented:
At least the workaround is great.  I had a broken production integration for a week.

Sent from my Windows Phone

---

From: Erik Selinmailto:notifications@github.com
Sent: ý6/ý15/ý2013 10:26 AM
To: jorgevaldivia/salesforce_bulkmailto:salesforce_bulk@noreply.github.com
Cc: Jamie Ayermailto:jayer@ioniasolutions.com
Subject: Re: [salesforce_bulk] Get rid of -api in regular expression for SalesforceBulk::Connection.parse_instance() (#16)

Just came to open a pr for the exact same fix but I see you guys beat me to it. Ping @jorgevaldiviahttps://github.com/jorgevaldivia as the gem is broken until this is fixed :(

—
Reply to this email directly or view it on GitHubhttps://github.com/jorgevaldivia/salesforce_bulk/pull/16#issuecomment-19497193.

Align-Deployment committed:
add Salesforce API error handling for salesforce_bulk gem;

Salesforce bulk API isn't so descriptive in error messages as simple
salesforce API.
These changes add ability to run failed query through
databasedotcom-rails gem and raise error with informative description
about what's going on.

Align-Deployment committed:
error handling should be only in read-only mode;

mahzilla committed:
Merge pull request #1 from Proffard/better-error-handling

add Salesforce API error handling for salesforce_bulk gem;

Commit:
Salesforce changed the configuration again…

Commit:
apply Proffard's changes

kevmimcc commented:
Any progress on this?  2 years later and the gem is still in a broken state?

seanpd commented:
I'm pretty sure it's fixed.  I do this:

gem 'salesforce_bulk', '1.0.3'

...and took out my monkeypatch workaround and everything still works.

On Mon, Aug 31, 2015 at 12:41 PM, Kevin notifications@github.com wrote:

> Any progress on this? 2 years later and the gem is still in a broken state?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jorgevaldivia/salesforce_bulk/pull/16#issuecomment-136440609
> .

JamieAyer commented:
I added the following to my application.rb file.

module SalesforceBulk
 class Connection
   def parse_instance()
     m = /https:\/\/([a-z]{2,2}[0-9]{1,2})(-api|.)/.match(@server_url)
     return '' unless m
     @instance = m[1]
   end
 end
End

Jamie Ayer  |   Solution Architect
320 Summer Street, STE 100, Boston, MA 02210-1701 USA
Xively by LogMeInhttps://xively.com/  |  O. +1-781-247-2858  |  M. +1-617-413-3351

[Description: C:\Users\llyons\Desktop\Xively Marketing\Branding\Logos-Habanero\xively_by_LMI-right_habanero.png]https://xively.com/

From: Sean Dempsey <notifications@github.com<mailto:notifications@github.com>>
Reply-To: jorgevaldivia/salesforce_bulk <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Monday, August 31, 2015 at 1:47 PM
To: jorgevaldivia/salesforce_bulk <salesforce_bulk@noreply.github.com<mailto:salesforce_bulk@noreply.github.com>>
Cc: Jamie Ayer <Jamie.Ayer@logmein.com<mailto:Jamie.Ayer@logmein.com>>
Subject: Re: [salesforce_bulk] Get rid of -api in regular expression for SalesforceBulk::Connection.parse_instance() (#16)

I'm pretty sure it's fixed. I do this:

gem 'salesforce_bulk', '1.0.3'

...and took out my monkeypatch workaround and everything still works.

On Mon, Aug 31, 2015 at 12:41 PM, Kevin <notifications@github.com<mailto:notifications@github.com>> wrote:

> Any progress on this? 2 years later and the gem is still in a broken state?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jorgevaldivia/salesforce_bulk/pull/16#issuecomment-136440609
> .

—
Reply to this email directly or view it on GitHubhttps://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_jorgevaldivia_salesforce-5Fbulk_pull_16-23issuecomment-2D136441826&d=AwMFaQ&c=k6LEwkJiSX1kJxtgPu1uYQ&r=TD7G2BmSQlGI8YouSwiNkCDus3pab4GbU7cfWb_OoBo&m=QigdHe3Or-mdrvlyfi4PLZ6pwbQQY0jCQohYiMSdCDQ&s=pDRPwwBfJN5e5BiHsOjoYE8RiARn8ch4-xaJtC5UA6s&e=.
